plugins/org.eclipse.fordiac.ide.structuredtextcore.ui/src/org/eclipse/fordiac/ide/structuredtextcore/ui/resource/STCoreResourceForIEditorInputFactory.java

```diff
@@ -27,17 +27,15 @@ public class STCoreResourceForIEditorInputFactory extends ResourceForIEditorInpu
 	@Override
 	public Resource createResource(final IEditorInput editorInput) {
 		final Resource resource = super.createResource(editorInput);
-		if (resource instanceof final LibraryElementXtextResource libraryElementXtextResource) {
-			libraryElementXtextResource.setEagerLinking(true);
-			if (editorInput instanceof final IFileEditorInput fileEditorInput) {
-				final TypeEntry typeEntry = TypeLibraryManager.INSTANCE.getTypeEntryForFile(fileEditorInput.getFile());
-				if (typeEntry != null) {
-					final LibraryElement libraryElement = typeEntry.getTypeEditable();
-					libraryElementXtextResource.setLibraryElement(libraryElement);
-					libraryElementXtextResource.setIncludeInternalLibraryElement(libraryElement instanceof BaseFBType);
-					if (resource instanceof final STCoreResource stCoreResource) {
-						stCoreResource.getDefaultLoadOptions().put(STCoreResource.OPTION_PLAIN_ST, Boolean.TRUE);
-					}
+		if (resource instanceof final LibraryElementXtextResource libraryElementXtextResource
+				&& editorInput instanceof final IFileEditorInput fileEditorInput) {
+			final TypeEntry typeEntry = TypeLibraryManager.INSTANCE.getTypeEntryForFile(fileEditorInput.getFile());
+			if (typeEntry != null) {
+				final LibraryElement libraryElement = typeEntry.getTypeEditable();
```

Comment:
Is this method always called within a typeeditor. If yes it would be better to check for a TypeEditorInput and get the editable libraryElement from there. This avoids duplicated loads and potential mismatches between the editor pages.

Reply:
Yes, I was planning that for a later PR, including other places in the ST editor that currently use `getTypeEditable()`.

Reply:
Ok good to know